src/coreclr/jit/lsraarm64.cpp

```diff
@@ -25,6 +25,65 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "sideeffects.h"
 #include "lower.h"
 
+//------------------------------------------------------------------------
+// getNextConsecutiveRefPosition: Get the next subsequent refPosition.
+//
+// Arguments:
+//    refPosition   - The refposition for which we need to find next refposition
+//
+// Return Value:
+//    The next refPosition or nullptr if there is not one.
+//
+RefPosition* LinearScan::getNextConsecutiveRefPosition(RefPosition* refPosition)
+{
+    RefPosition* nextRefPosition;
+    assert(refPosition->needsConsecutive);
+    nextConsecutiveRefPositionMap->Lookup(refPosition, &nextRefPosition);
+    assert((nextRefPosition == nullptr) || nextRefPosition->needsConsecutive);
+    return nextRefPosition;
+}
+
+//------------------------------------------------------------------------
+// setNextConsecutiveRegisterAssignment: Set the consecutive register mask to the
+//   subsequent refpositions
+//
+// Arguments:
+//    tree       - The GT_HWINTRINSIC node of interest
+//    pDstCount  - OUT parameter - the number of registers defined for the given node
+//
+// Return Value:
+//    The number of sources consumed by this node.
+//
+void LinearScan::setNextConsecutiveRegisterAssignment(RefPosition* firstRefPosition, regMaskTP firstRegAssigned)
+{
+    assert(isSingleRegister(firstRegAssigned));
+    assert(firstRefPosition->isFirstRefPositionOfConsecutiveRegisters());
+
+    RefPosition* consecutiveRefPosition = getNextConsecutiveRefPosition(firstRefPosition);
+
+    // should have at least one consecutive register requirement
+    assert(consecutiveRefPosition != nullptr);
+
+    regNumber firstReg    = genRegNumFromMask(firstRegAssigned);
+    regNumber regToAssign = firstReg == REG_FP_LAST ? REG_FP_FIRST : REG_NEXT(firstReg);
+
+#ifdef DEBUG
+    int refPosCount = 1;
```

Comment:
There's an `INDEBUG` macro that can help trim down on `#ifdef DEBUG` code volume:
```
INDEBUG(int refPosCount = 1);
...
INDEBUG(refPosCount++);
```
If you like that kind of thing.